Fail to build in dev container

### Report

I tried building the project in dev container as instructed in [BUILD.md](https://github.com/kedacore/keda/blob/main/BUILD.md#quick-start-with-visual-studio-code-remote---containers). 
As I ran the command `make build`, the build terminated due to some `undefined reference` errors.

### Expected Behavior

Seeing a successful build

### Actual Behavior

Receiving errors
```
root@dd666a615af0:/workspaces/keda# make build
test -s /workspaces/keda/bin/controller-gen || GOBIN=/workspaces/keda/bin go install sigs.k8s.io/controller-tools/cmd/controller-gen
test -s /workspaces/keda/bin/mockgen || GOBIN=/workspaces/keda/bin go install github.com/golang/mock/mockgen
PATH="/workspaces/keda/bin:/vscode/vscode-server/bin/linux-arm64/8b3775030ed1a69b13e4f4c628c612102e30a681/bin/remote-cli:/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin" protoc -I vendor --proto_path=hack LiiklusService.proto --go_out=pkg/scalers/liiklus --go-grpc_out=pkg/scalers/liiklus
PATH="/workspaces/keda/bin:/vscode/vscode-server/bin/linux-arm64/8b3775030ed1a69b13e4f4c628c612102e30a681/bin/remote-cli:/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin" protoc -I vendor --proto_path=pkg/scalers/externalscaler externalscaler.proto --go_out=pkg/scalers/externalscaler --go-grpc_out=pkg/scalers/externalscaler
PATH="/workspaces/keda/bin:/vscode/vscode-server/bin/linux-arm64/8b3775030ed1a69b13e4f4c628c612102e30a681/bin/remote-cli:/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin" protoc -I vendor --proto_path=pkg/metricsservice/api metrics.proto --go_out=pkg/metricsservice/api --go-grpc_out=pkg/metricsservice/api
/workspaces/keda/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
go fmt ./...
go vet ./...
# github.com/DataDog/datadog-api-client-go/api/v1/datadog
vendor/github.com/DataDog/datadog-api-client-go/api/v1/datadog/client.go:100:11: undefined: TagsApiService
vendor/github.com/DataDog/datadog-api-client-go/api/v1/datadog/model_formula_and_function_metric_query_definition.go:17:13: undefined: FormulaAndFunctionMetricDataSource
vendor/github.com/DataDog/datadog-api-client-go/api/v1/datadog/model_event_timeline_widget_definition.go:27:7: undefined: EventTimelineWidgetDefinitionType
vendor/github.com/DataDog/datadog-api-client-go/api/v1/datadog/model_free_text_widget_definition.go:23:7: undefined: FreeTextWidgetDefinitionType
vendor/github.com/DataDog/datadog-api-client-go/api/v1/datadog/model_table_widget_definition.go:29:7: undefined: TableWidgetDefinitionType
vendor/github.com/DataDog/datadog-api-client-go/api/v1/datadog/model_widget_request_style.go:14:12: undefined: WidgetLineType
vendor/github.com/DataDog/datadog-api-client-go/api/v1/datadog/model_widget_definition.go:39:34: undefined: TreeMapWidgetDefinition
vendor/github.com/DataDog/datadog-api-client-go/api/v1/datadog/model_funnel_query.go:19:10: undefined: FunnelStep
vendor/github.com/DataDog/datadog-api-client-go/api/v1/datadog/model_logs_list_request.go:30:7: undefined: LogsListRequestTime
vendor/github.com/DataDog/datadog-api-client-go/api/v1/datadog/model_logs_exclusion.go:15:10: undefined: LogsExclusionFilter
vendor/github.com/DataDog/datadog-api-client-go/api/v1/datadog/model_logs_exclusion.go:15:10: too many errors
```

### Steps to Reproduce the Problem

1. 
```
git clone git@github.com:kedacore/keda.git
cd keda
code .
```
2. run CTRL+SHIFT+P -> Remote-Containers: Reopen in container
3. use the integrated terminal to run: `make build`

### Logs from KEDA operator

```
example
```


### KEDA Version

2.13.0

### Kubernetes Version

None

### Platform

None

### Scaler Details

_No response_

### Anything else?

_No response_

--- a/Makefile
+++ b/Makefile
@@ -188,7 +188,11 @@ pkg/scalers/liiklus/mocks/mock_liiklus.go:
 
 ##@ Build
 
-build: generate fmt vet manager adapter webhooks ## Build Operator (manager), Metrics Server (adapter) and Admision Web Hooks (webhooks) binaries.
+build: update-mod generate fmt vet manager adapter webhooks ## Build Operator (manager), Metrics Server (adapter) and Admision Web Hooks (webhooks) binaries.
+
+update-mod:
+	go mod tidy
+	go mod vendor
 
 manager: generate
 	${GO_BUILD_VARS} go build -ldflags $(GO_LDFLAGS) -mod=vendor -o bin/keda cmd/operator/main.go

